lib/resqued/quit-and-wait.rb

```diff
@@ -0,0 +1,83 @@
+require "optparse"
+
+module Resqued
+  class QuitAndWait
+    def self.exec!(argv)
+      options = {grace_seconds: 30}
+
+      opts = OptionParser.new do |opts|
+        opts.banner = <<USAGE
+Usage: resqued quit-and-wait PIDFILE [--grace-period SECONDS]
+
+Use this as a preStop script in kubernetes. This script will send a SIGQUIT to
+resqued immediately, and then sleep until either resqued exits or until the
+grace period is approximately 5 seconds from expiration. This script exits 0 if
+resqued exited and 99 otherwise.
+USAGE
+
+        opts.on "-h", "--help", "Show this message" do
+          puts opts
+          exit
+        end
+
+        opts.on "-v", "--version", "Show the version" do
+          require "resqued/version"
+          puts Resqued::VERSION
+          exit
+        end
+
+        opts.on "--grace-period SECONDS", Numeric, "If resqued does not exit within SECONDS (default 15) seconds, exit with an error" do |v|
```

Comment:
Should this be 25?
```suggestion
        opts.on "--grace-period SECONDS", Numeric, "If resqued does not exit within SECONDS (default 25) seconds, exit with an error" do |v|
```

Reply:
Good catch! I ended up revising the docs a bit more to try to make it clearer that this should match kube's terminationGracePeriod, rather than directly defining the timeout.